### PR TITLE
refactor: Optimize aeskeygenassist_si128 in ARM64

### PR DESCRIPTION
**Pull request by howjmay**
Conduct VTRN to remove bit selection operations.

### CODE REVIEW

**Review by jserv — CHANGES_REQUESTED** [review]
Refine the subject of this pull request and git commit message. You shall emphasize the benefit with the proposed changes.

File: sse2neon.h [hunk 1]
```diff
@@ -9906,19 +9906,80 @@ FORCE_INLINE __m128i _mm_aesimc_si128(__m128i a)
...
-        u8[0x1], u8[0xE], u8[0xB], u8[0x4],  // ROT(SubBytes(X1))
-        u8[0xC], u8[0x9], u8[0x6], u8[0x3],  // SubBytes(X3)
-        u8[0x9], u8[0x6], u8[0x3], u8[0xC],  // ROT(SubBytes(X3))
-    };
-    uint32x4_t r = {0, (unsigned) rcon, 0, (unsigned) rcon};
-    return vreinterpretq_m128i_u8(dest) ^ vreinterpretq_m128i_u32(r);
+#define SSE2NEON_AES_SBOX(w)                                               \
+    {                                                                      \
+        w(0x63), w(0x7c), w(0x77), w(0x7b), w(0xf2), w(0x6b), w(0x6f),     \
+            w(0xc5), w(0x30), w(0x01), w(0x67), w(0x2b), w(0xfe), w(0xd7), \
```
Comment (jserv) [inline sse2neon.h:9912]: Indent properly.

File: sse2neon.h [hunk 2]
```diff
@@ -9906,19 +9906,80 @@ FORCE_INLINE __m128i _mm_aesimc_si128(__m128i a)
...
-    uint8x16_t dest = {
-        // Undo ShiftRows step from AESE and extract X1 and X3
-        u8[0x4], u8[0x1], u8[0xE], u8[0xB],  // SubBytes(X1)
-        u8[0x1], u8[0xE], u8[0xB], u8[0x4],  // ROT(SubBytes(X1))
-        u8[0xC], u8[0x9], u8[0x6], u8[0x3],  // SubBytes(X3)
-        u8[0x9], u8[0x6], u8[0x3], u8[0xC],  // ROT(SubBytes(X3))
-    };
-    uint32x4_t r = {0, (unsigned) rcon, 0, (unsigned) rcon};
-    return vreinterpretq_m128i_u8(dest) ^ vreinterpretq_m128i_u32(r);
+#define SSE2NEON_AES_SBOX(w)                                               \
```
Comment (jserv) [inline sse2neon.h:9909]: Can you move the definition out of function body?

Comment (jserv) [reply sse2neon.h:9909]: Isn't `SSE2NEON_AES_SBOX` existing somewhere in the header already?

**Comment by howjmay** [comment]
I found the previous implementation actually have used intrinsics. I will check the performance of my current implementation first, then reopen this PR.

**Pull request merged**